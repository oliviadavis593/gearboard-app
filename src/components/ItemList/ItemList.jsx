import React, { Component } from 'react';
import GearContext from '../../GearContext'
import Item from '../Item/Item';


class ItemList extends Component {
    static contextType = GearContext; 
    render() {

        const items = this.context.items; 

        return(
            <section className='gb-item-list__container'>
                <h2>Gear List</h2>
                <ul className='gb-item_list'>
                    {items.map(item => 
                        <Item 
                        key={item.id}
                        {...item}
                        />    
                    )}
                </ul>
            </section>
        )
    }
}

export default ItemList; 