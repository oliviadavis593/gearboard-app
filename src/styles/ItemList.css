

.gb-item_list{
    list-style-type: none; 
    text-align: center; 
    margin: 0 auto; 
}

.gb-item__delete {
    --item-button-height: 32px;
    background-color: #cf6d17;
    border: 0;
    color: #eee;
    font-size: 12px;
    height: var(--item-button-height);
    top: calc((var(--item-button-height) + 80px) * -1);
    transform-origin: right bottom;
    width: calc(var(--item-height)); 
}