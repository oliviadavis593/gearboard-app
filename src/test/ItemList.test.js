import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import ItemList from '../components/ItemList/ItemList'

describe('ItemList Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ItemList />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ItemList gear_name='gear_name' id={2}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})