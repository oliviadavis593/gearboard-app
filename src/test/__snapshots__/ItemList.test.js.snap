// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Component renders the UI as expected 1`] = `
<section
  className="gb-item-list__container"
>
  <ul
    className="gb-item_list"
  />
</section>
`;
